client/src/components/Draftail/CommentableEditor/CommentableEditor.tsx

```diff
@@ -507,30 +540,10 @@ function CommentableEditor({
 
   useEffect(() => {
     // if there are any comments without annotations, we need to add them to the EditorState
-    let contentState = editorState.getCurrentContent();
-    let hasUpdated = false;
-    comments.filter(comment => !comment.annotation).forEach((comment) => {
-      commentApp.updateAnnotation(new DraftailInlineAnnotation(fieldNode), comment.localId);
-      const style = `${COMMENT_STYLE_IDENTIFIER}${comment.localId}`;
-      try {
-        const positions = JSON.parse(comment.position);
-        positions.forEach((position) => {
-          contentState = applyInlineStyleToRange({
-            contentState,
-            blockKey: position.key,
-            start: position.start,
-            end: position.end,
-            style
-          });
-          hasUpdated = true;
-        });
-      } catch (err) {
-        console.error(`Error loading comment position for comment ${comment.localId}`);
-        console.error(err);
-      }
-    });
-    if (hasUpdated) {
-      setEditorState(forceResetEditorState(editorState, contentState));
+    const contentState = editorState.getCurrentContent();
```

Comment:
This might be why I get Draftail messages every time I type a character in a comment. This code is probably being run for every comment change, not just ones that are relevant for Draftail

Reply:
This particular bit should only run:
- For comments with a relevant draftail contentpath
- Once per comment, as it runs for comments with no annotation and then adds an annotation
So I think it's unlikely to be the culprit

That said, I think if you're changing the comment text for a draftail comment, the other `useEffect` hook could potentially trigger to rerender Draftail, which we could probably fix by changing the id equality check to not work by reference only, so that could be the culprit